lib/puppet/functions/batch_escape.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+# @summary
+#   Escapes a string so that it can be safely used in a batch shell command line.
+#
+# >* Note:* that the resulting string should be used unquoted and is not intended for use in double quotes nor in single
+# quotes.
+Puppet::Functions.create_function(:batch_escape) do
+  # @return
```

Comment:
Should probably document the parameter here too, and regenerate the REFERENCE.MD to check everything looks good.

Reply:
Well, I copy-pasted the original comments and REFERENCE.md was unchanged so I was happy with it, but it looks like `bundle exec rake strings:generate:reference` in fact fails and the REFERENCE.md is not updated…  I am digging into this…

Reply:
I will track this issue in another PR.

Reply:
I updated #1239 to improve this and avoid it in the future :wink: